packages/api/src/external/carequality/patient.ts

```diff
@@ -28,29 +30,83 @@ import { PatientDataCarequality } from "./patient-shared";
 dayjs.extend(duration);
 
 const context = "cq.patient.discover";
-const iheGateway = makeIheGatewayAPIForPatientDiscovery();
 const resultPoller = makeOutboundResultPoller();
 
-export async function discover(patient: Patient, facilityNPI: string): Promise<void> {
+export async function discover(
+  patient: Patient,
+  facilityNPI: string,
+  forceEnabled = false
+): Promise<void> {
   const baseLogMessage = `CQ PD - patientId ${patient.id}`;
   const { log: outerLog } = out(baseLogMessage);
   const { cxId } = patient;
 
-  if (!iheGateway) return outerLog(`IHE GW not available, skipping PD`);
-  if (!(await isCQDirectEnabledForCx(cxId))) {
-    return outerLog(`CQ disabled for cx ${cxId}, skipping PD`);
+  const enabledIHEGW = await validatePDEnabledAndInitGW(cxId, forceEnabled, outerLog);
+
+  if (enabledIHEGW) {
+    await processPatientDiscoveryProgress({ patient, status: "processing" });
+
+    // Intentionally asynchronous
+    prepareAndTriggerPD(patient, facilityNPI, enabledIHEGW, baseLogMessage).catch(
```

Comment:
This doesn't need to be asynchronous here, as `discover` is already called in an async manner

Reply:
oh wait, no, it was moved do synchronous... ok then